packages/relay/src/lib/eth.ts

```diff
@@ -483,6 +484,11 @@ export class EthImpl implements Eth {
       }
     } catch (e: any) {
       this.logger.error(`${requestIdPrefix} Error raised while fetching estimateGas from mirror-node: ${JSON.stringify(e)}`);
+      
+      // execute reverted: with no message comes from a contract deployment
+      if ((e.statusCode != 501) && (e instanceof JsonRpcError) && (e.message === EthImpl.executionReverted)) {
```

Comment:
I don't think this will work, the error here if it's a contract revert is not an HTTPError. 
`e.statusCode != 501` will be true for other cases and even when the mirror node returns 400 it won't make it here as the same error. 
`JSONRpcError` doesn't even have a `statusCode` field

Look at [MirrorNodeClient.handleError()](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/clients/mirrorNodeClient.ts#L335), it throws a `predefined.CONTRACT_REVERT(mirrorError.errorMessage)`.
Which means here the error is a new error type. I think the suggestion i provided previously is closer to what you need.

```suggestion
      if ((e.statusCode == MirrorNodeClientError.ErrorCodes) && (e instanceof JsonRpcError) && (e.message === EthImpl.executionReverted)) {
```

Reply:
Merged in the latest mirror node error handling fix and updated the condition to:
```
        if (e.message === EthImpl.errorContractReverted) {
          return predefined.CONTRACT_REVERT(e.detail, e.data);
        }      
```